src/shell.rs

```diff
@@ -78,4 +129,76 @@ mod tests {
                                      .join(".bashrc");
         assert_eq!(s.path_for(".bashrc"), expected_path);
     }
+
+    #[test]
+    fn can_link_files() {
+        let test_root = set_up("shell-link");
+    
+        fs::create_dir(test_root.join(".config"));
+        fs::create_dir(test_root.join(".config/hermit"));
+        fs::create_dir(test_root.join(".config/hermit/shells"));
+        fs::create_dir(test_root.join(".config/hermit/shells/default"));
+
+        let shell_root = test_root.join(".config/hermit/shells/default");
+        fs::File::create(shell_root.join(".bashrc"));
+        fs::File::create(shell_root.join(".gitconfig"));
+        fs::create_dir(shell_root.join("foo"));
+        fs::File::create(shell_root.join("foo/bar.txt"));
```

Comment:
Can we pull this filesystem setup out into a function for naming purposes? This test is quite long.